Marlin/src/HAL/ESP32/HAL_SPI.cpp

```diff
@@ -53,11 +53,13 @@ static SPISettings spiConfig;
 // ------------------------
 
 void spiBegin() {
-  #if !PIN_EXISTS(SD_SS)
-    #error "SD_SS_PIN not defined!"
-  #endif
+  #if ENABLED(SDSUPPORT)  
+    #if !PIN_EXISTS(SD_SS)
+      #error "SD_SS_PIN not defined!"
+    #endif
 
-  OUT_WRITE(SD_SS_PIN, HIGH);
+    OUT_WRITE(SD_SS_PIN, HIGH);
+  #endif
```

Comment:
In case this seems weird, I guess the background is this: Just about every board has `SD_SS_PIN` so this is the minimal initialization of the SPI bus to make sure there are no active devices at startup. The casual reader might forget that CS is active-low, so we might want to add comments to all the SPI code that touches SS/CS pins as a reminder.